Python/pystate.c

```diff
@@ -1255,10 +1255,14 @@ PyFrameObject*
 PyThreadState_GetFrame(PyThreadState *tstate)
 {
     assert(tstate != NULL);
-    if (tstate->cframe->current_frame == NULL) {
+    _PyInterpreterFrame *f = tstate->cframe->current_frame;
+    while (f && _PyFrame_IsIncomplete(f)) {
+        f = f->previous;
```

Comment:
Is `previous` guaranteed to be initialized in an incomplete frame?

Reply:
Yes. A `_PyInterpreterFrame` is never incomplete in the sense that it is partially initialized. "Incomplete" just means that the resulting frame object wouldn't be valid because stuff like `MAKE_CELL` hasn't run yet (or we just want to hide the frame).